server/src/main/java/org/opensearch/action/pagination/WlmPaginationStrategy.java

```diff
@@ -0,0 +1,193 @@
+import org.opensearch.wlm.stats.WlmStats;
+import org.opensearch.wlm.stats.QueryGroupStats;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+
+/**
+ * Pagination strategy for Workload Management (WLM) Stats.
+ * Paginates based on query group IDs.
+ */
+public class WlmPaginationStrategy implements PaginationStrategy<WlmStats> {
+    private static final String DEFAULT_PAGINATED_ENTITY = "wlm_stats";
+
+    private final int pageSize;
+    private final String nextToken;
+    private final String sortBy;
+    private final String sortOrder;
+    private final List<WlmStats> paginatedStats;
+    private PageToken responseToken;
+
+    public WlmPaginationStrategy(int pageSize, String nextToken, String sortBy, String sortOrder, WlmStatsResponse response) {
+        this.pageSize = pageSize;
+        this.nextToken = nextToken;
+        this.sortBy = sortBy;
+        this.sortOrder = sortOrder;
+
+        WlmStrategyToken requestedToken = (nextToken == null || nextToken.isEmpty())
+            ? null
+            : new WlmStrategyToken(nextToken);
+
+        List<WlmStats> sortedStats = sortStats(response.getNodes(), sortBy, sortOrder);
+        this.paginatedStats = applyPagination(sortedStats, requestedToken);
+    }
+
+    /**
+     * Sorts WlmStats based on the provided attribute (`node_id` or `query_group`).
+     */
+    private List<WlmStats> sortStats(List<WlmStats> stats, String sortBy, String sortOrder) {
+        Comparator<WlmStats> comparator;
+
+        switch (sortBy) {
+            case "query_group":
+                comparator = Comparator.comparing(
+                    (WlmStats wlmStats) -> wlmStats.getQueryGroupStats().getStats().isEmpty()
+                        ? ""
+                        : wlmStats.getQueryGroupStats().getStats().keySet().iterator().next()
+                ).thenComparing(wlmStats -> wlmStats.getNode().getId());
+                break;
+
+            case "node_id":  // Default sorting
+            default:
+                comparator = Comparator.comparing(
+                    (WlmStats wlmStats) -> wlmStats.getNode().getId()
+                ).thenComparing(
+                    wlmStats -> wlmStats.getQueryGroupStats().getStats().isEmpty()
+                        ? ""
+                        : wlmStats.getQueryGroupStats().getStats().keySet().iterator().next()
+                );
+                break;
+        }
+
+        if ("desc".equals(sortOrder)) {
+            comparator = comparator.reversed();
+        }
+
+        return stats.stream().sorted(comparator).collect(Collectors.toList());
+    }
+
+    /**
+     * Applies pagination on already sorted WlmStats.
+     */
+    private List<WlmStats> applyPagination(List<WlmStats> sortedStats, WlmStrategyToken requestedToken) {
```

Comment:
We should also check for transient updates to query group count in the cluster during subsequent calls to the API. For example lets say queryGroupCount was 2 when you made first call to the stats API but in between during first and second call to stats API there was an addition to the QueryGroups then it will return inconsistent results.